utils.js

```diff
@@ -26,4 +26,22 @@ function onceWhen(obv, filter, cb) {
   })
 }
 
-module.exports = { onceWhen }
+class ReadyGate {
```

Comment:
I made this class because I wanted "synchronous deferred promises", but no such thing exists. The internals of this class may be familiar to you, we use this kind of technique in jitdb: https://github.com/ssb-ngi-pointer/jitdb/blob/8f3b79abaf2c90b673323f461dd9abd3910a0a88/index.js#L41-L42